src/client/opamArg.ml

```diff
@@ -38,22 +38,18 @@ type global_options = {
   cli : OpamCLIVersion.t;
 }
 
-let deprecated_option option absent name instead =
-  if option <> absent then
-    OpamConsole.warning
-      "Option %s is deprecated, ignoring it.%s" name
-      (match instead with
-      | None -> ""
-      | Some instead -> Printf.sprintf " You can use %s instead." instead)
-
 (* The --cli passed by cmdliner is ignored (it's only there for --help) *)
 let create_global_options
     git_version debug debug_level verbose quiet color opt_switch yes strict
     opt_root external_solver use_internal_solver
     cudf_file solver_preferences best_effort safe_mode json no_auto_upgrade
     working_dir ignore_pin_depends
     d_no_aspcud _ =
-  deprecated_option d_no_aspcud false "no-aspcud" None;
+  if d_no_aspcud then
+    OpamConsole.warning
+      "Option %s is deprecated, ignoring it. \
+       Use instead 'opam switch <switch-name> <compiler>'"
```

Comment:
This message isn't correct! There was no extra message before.

Reply:
There was, in the [`deprecated_option`](https://github.com/ocaml/opam/pull/4385/files#diff-e674376a5ba688da9b6f11571be146af2e7bbde7c583630c6e8d4fc15c575035L41) function

Reply:
There was an optional parameter, but it wasn't used for _this_ option. opam 2.0 just gives:

```
[WARNING] Option no-aspcud is deprecated, ignoring it.
```

Reply:
oups, bad copy-pasting. I thought you were talking globally, not for this message particularly.